### ISSUE TITLE
Another app is currently holding the xtables lock

### ISSUE THREAD

**Issue by madchap** [sshuttle/sshuttle#214]
Hi,

Running opensuse tumbleweed, and never had any issues until a few weeks back.

I now get error messages from sshuttle, such as:

```
Another app is currently holding the xtables lock. Perhaps you want to use the -w option?
Another app is currently holding the xtables lock. Perhaps you want to use the -w option?
Another app is currently holding the xtables lock. Perhaps you want to use the -w option?
error: ['iptables', '-t', 'nat', '-D', 'OUTPUT', '-j', 'sshuttle-12293'] returned 4
iptables: No chain/target/match by that name.
Another app is currently holding the xtables lock. Perhaps you want to use the -w option?
error: ['iptables', '-t', 'nat', '-D', 'PREROUTING', '-j', 'sshuttle-12293'] returned 1
error: ['iptables', '-t', 'nat', '-D', 'OUTPUT', '-j', 'sshuttle-12300'] returned 4
Another app is currently holding the xtables lock. Perhaps you want to use the -w option?
error: ['iptables', '-t', 'nat', '-D', 'PREROUTING', '-j', 'sshuttle-12300'] returned 4
fatal: ['iptables', '-t', 'nat', '-I', 'OUTPUT', '1', '-j', 'sshuttle-12293'] returned 4
fatal: ['iptables', '-t', 'nat', '-F', 'sshuttle-12300'] returned 4
client: fatal: cleanup: ['sudo', '-p', '[local sudo] Password: ', 'PYTHONPATH=/usr/lib/python2.7/site-packages', '--', '/usr/bin/python', '/usr/bin/sshuttle', '--method', 'auto', '--firewall'] returned 99
client: fatal: cleanup: ['sudo', '-p', '[local sudo] Password: ', 'PYTHONPATH=/usr/lib/python2.7/site-packages', '--', '/usr/bin/python', '/usr/bin/sshuttle', '--method', 'auto', '--firewall'] returned 99
client: Connected.
client: Connected.
Another app is currently holding the xtables lock. Perhaps you want to use the -w option?
Another app is currently holding the xtables lock. Perhaps you want to use the -w option?
error: ['iptables', '-t', 'nat', '-D', 'PREROUTING', '-j', 'sshuttle-12299'] returned 4
``` 

The "client: Connected" strings there are just an ephemeral state, as the connection just dies quickly afterwards. I run several "sshuttle" processes, like 1 per subnet basically.

firewalld is running, none of the SuSEfirewall2 stuff.

Thanks for any insight. This is a massive bummer in my current workflow.
Let me know if I can provide any other info.

Cheers,
fred

**Comment by madchap** [comment]
$ rpm -q firewalld                                                                             
firewalld-0.5.1-1.1.noarch

 $ rpm -q sshuttle                                                                              
sshuttle-0.78.1-2.3.noarch

**Comment by madchap** [comment]
I would be tempted to say it would potentially coincides with this: https://lists.opensuse.org/opensuse-factory/2018-01/msg00643.html

**Comment by madchap** [comment]
I reverted to the latest firewalld package from opensuse LEAP, and it seems to work.

```
$ rpm -q firewalld
firewalld-0.4.4.5-1.1.noarch
```

So, definitely something with it then..

**Comment by madchap** [comment]
Opened bug on opensuse's bugzilla as well: https://bugzilla.opensuse.org/show_bug.cgi?id=1085697

**Comment by brianmay** [comment]
See: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=816087

Possibly passing `-w` to iptables might help. However this to me sounds more like "firewalld keeps the tables locked constantly while it is running" as opposed to "a race condition between multiple processes updating the firewall tables" in which case this won't help.

**Comment by madchap** [comment]
mokay. I stopped firewalld entirely... and I have the same issue. I guess the problem is not there.

**Comment by madchap** [comment]
Well, maybe I got lucky for all these long months, having sshuttle work the way I had it to, and some collateral updates screw me over a couple months ago.

I had a script run sshuttle multiple times, once per subnet, so I can have a visual cue on how many I had connected, not using the `-D` switch, and rather doing the `&` at the end of the command.

I just tried to have one single sshuttle with all subnets in it, and it seems to work, with firewalld enabled.

I am closing this for now, and it very well may be an issue I had just created to myself. I will also close the other bug on opensuse.org.

Sorry for the noise.
